yazi-fs/src/provider/local/casefold.rs

```diff
@@ -123,6 +122,11 @@ fn final_path(path: &Path) -> io::Result<PathBuf> {
 	Ok(OsString::from_vec(cstr.to_bytes().to_vec()).into())
 }
 
+#[cfg(target_os = "openbsd")]
+fn final_path(path: &Path) -> io::Result<PathBuf> {
+    std::fs::canonicalize(path)
```

Comment:
This is not right as `final_path` should not follow symlinks but `canonicalize` does

Reply:
component-based normalization won't work in this case unfortunately since `final_path` needs to obtain the actual path of the given path on the filesystem without following symlinks, which means that when passed `/root/abc`, if the filesystem is case‑insensitive and the real entry is named `ABC`, it should return `/root/ABC` instead.